cmd/panacead/cmd/genoracle.go

```diff
@@ -0,0 +1,225 @@
+		Short: "Add a genesis oracle and oracle module parameters to genesis.json",
+		Long: `
+			Using this command, you can set genesis oracle and oracle module parameters.
+			If you set oracle public key path for oracle module params, the flags for oracle public key and remote report would be ignored.
+
+			The desired format of oracle public key file is:
+
+			"public_key_base64" : <base64-encoded-oracle-public-key>
+			"remote_report_base64" : <base64-encoded-oracle-remote-report>
+		`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			cdc := clientCtx.Codec
+
+			serverCtx := server.GetServerContextFromCmd(cmd)
+			config := serverCtx.Config
+
+			config.SetRoot(clientCtx.HomeDir)
+
+			genFile := config.GenesisFile()
+			appState, genDoc, err := genutiltypes.GenesisStateFromGenFile(genFile)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal genesis state: %w", err)
+			}
+
+			oracleGenState := oracletypes.GetGenesisStateFromAppState(cdc, appState)
+
+			if err := setOracle(cmd, oracleGenState); err != nil {
+				return fmt.Errorf("failed to set oracle: %w", err)
+			}
+
+			if err := setOracleParams(cmd, oracleGenState); err != nil {
+				return fmt.Errorf("failed to set oracle params: %w", err)
+			}
+
+			oracleGenStateBz, err := cdc.MarshalJSON(oracleGenState)
+			if err != nil {
+				return fmt.Errorf("failed to marshal oracle genesis state: %w", err)
+			}
+
+			appState[oracletypes.ModuleName] = oracleGenStateBz
+
+			appStateJSON, err := json.Marshal(appState)
+			if err != nil {
+				return fmt.Errorf("failed to marshal application genesis state: %w", err)
+			}
+
+			genDoc.AppState = appStateJSON
+
+			return genutil.ExportGenesisFile(genDoc, genFile)
+		},
+	}
+	cmd.Flags().String(flags.FlagKeyringBackend, flags.DefaultKeyringBackend, "Select keyring's backend (os|file|kwallet|pass|test)")
+	cmd.Flags().String(flagOracleUniqueID, "", "unique ID for oracle")
+	cmd.Flags().String(flagOraclePublicKey, "", "base64 encoded oracle public key")
+	cmd.Flags().String(flagOracleRemoteReport, "", "base64 encoded remote report of oracle public key")
+	cmd.Flags().String(flagOraclePublicKeyPath, "", "File path where 'oraclePublicKey' and 'oraclePublicKeyRemoteReport' are stored")
+	cmd.Flags().String(flagOracleAccount, "", "address or keyName")
+	cmd.Flags().String(flagOracleEndpoint, "", "oracle's endpoint")
+	cmd.Flags().String(flagOracleCommRate, "", "oracle's commission rate")
+	cmd.Flags().String(flags.FlagHome, defaultNodeHome, "The application home directory")
+
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func setOracle(cmd *cobra.Command, genState *oracletypes.GenesisState) error {
+	clientCtx := client.GetClientContextFromCmd(cmd)
+
+	oracleAddressOrKey, err := cmd.Flags().GetString(flagOracleAccount)
+	if err != nil {
+		return fmt.Errorf("failed to get oracle account flag: %w", err)
+	}
+
+	if len(oracleAddressOrKey) > 0 {
```

Comment:
This code would be much readable if you escape this function asap.
```suggestion
	if len(oracleAddressOrKey) == 0 {
	    return
	}
```
, or returning an error. I'm not sure which way is correct.